src/v/cluster/topics_frontend.cc

```diff
@@ -911,6 +918,18 @@ ss::future<std::vector<topic_result>> topics_frontend::autocreate_topics(
       leader.value(), std::move(topics), timeout);
 }
 
+ss::future<errc> topics_frontend::create_migrated_topic(
+  topic_configuration cfg, model::timeout_clock::time_point deadline) {
+    std::ignore = co_await stm_linearizable_barrier(deadline);
+
+    auto result = co_await do_create_topic(
+      custom_assignable_topic_configuration(std::move(cfg)), deadline);
+    if (result.ec == errc::success) {
+        std::ignore = co_await stm_linearizable_barrier(deadline);
+    }
```

Comment:
nit: it may be helpful to add a comment why we wait for barrier after creating a topic as it may be misleading

Reply:
I mimicked It from `topics_frontend::create_topics`. But ... :facepalm: ... this function is not in use. I'll drop it.